Exiled.API/Features/Doors/CheckpointDoor.cs

```diff
@@ -0,0 +1,81 @@
+// -----------------------------------------------------------------------
+// <copyright file="CheckpointDoor.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.API.Features.Doors
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Represents a check point door.
+    /// </summary>
+    public class CheckpointDoor : Door
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CheckpointDoor"/> class.
+        /// </summary>
+        /// <param name="door">The base <see cref="Interactables.Interobjects.CheckpointDoor"/> for this door.</param>
+        /// <param name="room">The <see cref="Room"/> for this door.</param>
+        public CheckpointDoor(Interactables.Interobjects.CheckpointDoor door, Room room)
+            : base(door, room)
+        {
+            Base = door;
+        }
+
+        /// <summary>
+        /// Gets the base <see cref="Interactables.Interobjects.CheckpointDoor"/>.
+        /// </summary>
+        public new Interactables.Interobjects.CheckpointDoor Base { get; }
+
+        /// <summary>
+        /// Gets the list of all sub doors for this <see cref="CheckpointDoor"/>.
+        /// </summary>
+        public IEnumerable<Door> Subdoors => Base.SubDoors.Select(Get);
```

Comment:
Make it load in the ctor
```suggestion
        public IReadOnlyCollection<Door> Subdoors { get; }
```